sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -256,28 +260,32 @@ private static Context setContextData(Span span) {
      */
     private void addSpanRequestAttributes(Span span, Context context, String spanName) {
         Objects.requireNonNull(span, "'span' cannot be null.");
-        span.setAttribute(
-            MESSAGE_BUS_DESTINATION,
-            AttributeValue.stringAttributeValue(getOrDefault(context, ENTITY_PATH_KEY, "", String.class)));
-        span.setAttribute(
-            PEER_ENDPOINT,
-            AttributeValue.stringAttributeValue(getOrDefault(context, HOST_NAME_KEY, "", String.class)));
-        span.setAttribute(
-            AZ_NAMESPACE_KEY,
-            AttributeValue.stringAttributeValue(getOrDefault(context, AZ_TRACING_NAMESPACE_KEY, "", String.class)));
+        String entityPath = getOrDefault(context, ENTITY_PATH_KEY, null, String.class);
+        if (entityPath != null) {
+            span.setAttribute(MESSAGE_BUS_DESTINATION, AttributeValue.stringAttributeValue(entityPath));
+        }
+        String hostName = getOrDefault(context, HOST_NAME_KEY, null, String.class);
+        if (hostName != null) {
+            span.setAttribute(PEER_ENDPOINT, AttributeValue.stringAttributeValue(hostName));
+        }
+        Long messageEnqueuedTime = getOrDefault(context, MESSAGE_ENQUEUED_TIME, null, Long.class);
```

Comment:
Should we convert time to `Instant` type? On the span, would a string formatted datetime make it easier than a long?

Reply:
There is a direct specification to have this as a long value. It could be directly be used as-is if kept long, maybe?
>Please add enqueuedTime attribute on each link when processing messages: unix epoch time with milliseconds precision representing when message was enqueued (x-opt-enqueued-time system property). Attribute value should have long type.